app/models/participant.rb

```diff
@@ -1,4 +1,7 @@
+# frozen_string_literal: true
+
 class Participant < ActiveRecord::Base
   belongs_to :project
+  has_secure_token :secret_id
```

Comment:
Making secret_id have a secure token is great as it auto-populates the field with a unique value for each participant and project.